### PR TITLE
feat!: migrate all curve backends to ff/group 0.14

### PR DESCRIPTION
**Pull request by mmaker**
Migrate every supported curve backend to ff/group 0.14: bls12_381 0.9, curve25519-dalek 5, and k256/p256 0.14. All existing default features remain available, including BLS12-381 G1 and G2. Update test and benchmark randomness to rand 0.10; the public prover API continues to use `ProverRng`.

The hash2group RFC 9380 implementations now use RustCrypto `hash2curve` 0.14 and the `shake` crate. K256 and P-256 use `LinearCombination::lincomb`; P-256 handles empty sums locally because primeorder 0.14 rejects them in debug builds. Its `alloc` feature is enabled explicitly so the optimized implementation is available when P-256 is the only backend enabled.

Retain BLS codecs, batched serialization, relation and batch-verification tests, specification vectors, and G1/G2 benchmarks. Extend MSM agreement checks to G2 and run timing smoke tests on BLS G1 so the generic multiplication implementation remains covered alongside the upgraded curve backends. Integration-test discovery stays enabled.

Keep main's existing Spongefish revision: codecs live in sigma-proofs, so this migration does not require switching to Spongefish 0.8. The declared Rust 1.88 minimum is unchanged.

### Validation

- Full workspace tests and doctests on the pinned toolchain and Rust 1.88; independent library builds for each curve backend on Rust 1.88.
- No-default-feature library and integration tests; BLS-only and P-256-only library tests; BLS-only batch-verification tests.
- Byte-exact BLS12-381 and P-256 specification vectors, including invalid proofs and contaminated batches; RFC 9380 expansion and hash-to-curve vectors.
- Default-feature workspace tests under Wasmtime.
- Release tests with overflow checks and debug assertions enabled; MSM benchmark smoke test.
- Clippy across workspace targets with all features and no default features; formatting and manifest sorting.
- Documentation with docs.rs flags and warnings treated as errors.
- Dependency advisory, license, source, and duplicate-version checks. All curve backends share one ff/group 0.14 dependency pair.

Supersedes #191 #194 #203 #205 #206 #208 #181 #172 #164 (and the closed #163).

### CODE REVIEW

File: src/linear_relation/canonical.rs [hunk 1]
```diff
@@ -255,13 +255,24 @@ impl<G: PrimeGroup> CanonicalLinearRelation<G> {
...
-    /// use hex_literal::hex;
+    /// # #[cfg(feature = "curve25519-dalek")] {
+    /// use curve25519_dalek::{RistrettoPoint, Scalar};
+    /// use group::Group;
+    /// use sigma_proofs::LinearRelation;
     /// use sigma_proofs::linear_relation::CanonicalLinearRelation;
-    /// type G = bls12_381::G1Projective;
+    /// type G = RistrettoPoint;
     ///
-    /// let dlog_instance_label = hex!("01000000000000000100000000000000010000009823a3def60a6e07fb25feb35f211ee2cbc9c130c1959514f5df6b5021a2b21a4c973630ec2090c733c1fe791834ce1197f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb");
```
Comment (nategraf) [inline src/linear_relation/canonical.rs:262]: This seems like a semantic change to the example. Is this intentional?

File: src/schnorr_protocol.rs [hunk 2]
```diff
@@ -194,12 +192,13 @@ where
     /// let p_var = relation.allocate_eq(x_var * g_var);
     ///
     /// relation.set_element(g_var, G::generator());
-    /// let x = Scalar::random(&mut OsRng);
+    /// let mut rng = rand::rng();
+    /// let x = Scalar::random(&mut rng);
```
Comment (nategraf) [inline src/schnorr_protocol.rs:196]:
```suggestion
    /// let x = Scalar::random(&mut rand::rng());
```

Comment (nategraf) [reply src/schnorr_protocol.rs:196]: It's possible this could simply be `Scalar::random(rand::rng())`, but I have not checked.

File: Cargo.toml [hunk 3]
```diff
@@ -18,6 +18,12 @@ categories.workspace = true
 keywords = ["cryptography", "zero-knowledge", "NIZK", "sigma-protocols"]
 description = "A toolkit for auto-generated implementations of Σ-protocols"
 exclude = [".gitignore"]
+# Test auto-discovery is disabled so that tests/test_relations.rs and
+# tests/test_batch_verification.rs (which require the temporarily removed
+# bls12_381 feature) stay in-tree without being built. Re-enable them with
+# explicit [[test]] sections once bls12_381 supports ff/group 0.14.
+# New integration tests must be registered under [[test]].
+autotests = false
```
Comment (nategraf) [inline Cargo.toml:26]: I would instead handle this by adding a `cfg` to the BLS tests files that makes them essentially empty.

**Review by nategraf — APPROVED** [review]

File: Cargo.toml (hunk 3 above)
Comment (mmaker) [reply Cargo.toml]: Addressed in 40047a0613559faa70a7d3e8997ee070f46e8346. The `autotests = false` setting is gone, so integration tests are discovered automatically. `test_relations` and `test_batch_verification` now run on Ristretto with their declared feature requirements. BLS-specific vector execution is deferred until a compatible backend returns, and the fixtures remain vendored. The relation and batch tests passed on both native and WASM builds.

File: src/schnorr_protocol.rs (hunk 2 above)
Comment (mmaker) [reply src/schnorr_protocol.rs]: `Scalar::random` takes as input `&mut Rng`, it looks like we can't compile without the borrow.
